daemon/src/maker_cfd_actor.rs

```diff
@@ -185,27 +197,77 @@ pub fn new(
+                                },
+                            },
+                            &mut conn,
+                        )
+                        .await
+                        .unwrap();
+                        cfd_feed_actor_inbox
+                            .send(load_all_cfds(&mut conn).await.unwrap())
+                            .unwrap();
+
                         tokio::spawn({
                             let sender = sender.clone();
 
                             async move {
                                 sender
-                                    .send(Command::CfdSetupCompleted(actor.await))
+                                    .send(Command::CfdSetupCompleted {
+                                        order_id,
+                                        dlc: actor.await,
+                                    })
                                     .unwrap()
                             }
                         });
-                        current_contract_setup = Some(inbox);
                     }
                     maker_cfd_actor::Command::IncProtocolMsg(msg) => {
                         let inbox = match &current_contract_setup {
-                            None => panic!("whoops"),
+                            None => {
+                                contract_setup_message_buffer.push(msg.clone());
+                                continue;
+                            }
                             Some(inbox) => inbox,
                         };
 
                         inbox.send(msg).unwrap();
                     }
-                    maker_cfd_actor::Command::CfdSetupCompleted(_finalized_cfd) => {
-                        todo!("but what?")
+                    maker_cfd_actor::Command::CfdSetupCompleted { order_id, dlc } => {
+                        dbg!("Setup complete, publishing on chain now...");
+
+                        current_contract_setup = None;
+                        contract_setup_message_buffer = vec![];
+
+                        insert_new_cfd_state_by_order_id(
+                            order_id,
+                            CfdState::PendingOpen {
+                                common: CfdStateCommon {
+                                    transition_timestamp: SystemTime::now(),
+                                },
+                                dlc,
+                            },
+                            &mut conn,
+                        )
+                        .await
+                        .unwrap();
+
+                        cfd_feed_actor_inbox
+                            .send(load_all_cfds(&mut conn).await.unwrap())
+                            .unwrap();
+
+                        // TODO: Publish on chain and only then transition to open - this might
+                        // require saving some internal state to make sure we are able to monitor
+                        // the publication after a restart
```

Comment:
Just try to publish on start up and handle error response accordingly would be my suggestion.

Reply:
Yeah, handling the state upon startup should be trivial. If we are in `PendingOpen we just use the same function to try to publish as here. Because we have to handle the case where it is already on chain anyway.